### PR TITLE
LMD GHOST fork choice implementation and tests

### PR DESCRIPTION
**Pull request by terenc3t**
This PR completes the frame work for LMD fork choice, it's not plugged in to block chain service (yet), I wanna do some more rigorously testing for longevity and performance before formally using it. 

Keep in mind there's room for optimization, V's implementation is not optimized, I don't foresee us using this native implementation long term. In the mean time, it'll be good to explore other implementations like Proto's

Key changes:
- Moved LMD fork choice to block chain service because it needs to recursively access beacon db to traverse through block DAGs
- Updated `VoteCount` and `BlockAncestor` to spec
- Implemented the following tests

**Test1**
```
B1 - B2 (vote:1)
```
Input: B1, Output: B2

**Test2**
```
       /- B2 (vote:1)
B1  - B3  (vote:1)
       \- B4`(vote:2)
```
Input: B1, Output: B4

**Test3**
```
       /- B2 - B4 - B6 (vote:1)
B1  - B3 - B5 (vote:2)
```
Input: B1, Output: B5

### CODE REVIEW

File: beacon-chain/blockchain/service_test.go [hunk 1]
```diff
@@ -1068,9 +1068,284 @@ func TestBlockChildren_SkipSlots(t *testing.T) {
...
+	// Construct the following chain:
+	//    /- B2
+	// B1  - B3 (State is slot 10)
+	//    \- B4
+	block1 := &pb.BeaconBlock{
+		Slot:             1,
+		ParentRootHash32: []byte{'A'},
+	}
+	root1, err := hashutil.HashBeaconBlock(block1)
+	if err != nil {
```
Comment (IvanTheGreatDev) [inline beacon-chain/blockchain/service_test.go]: Can this be changed to `if err := function(); err != nil {`?

File: beacon-chain/blockchain/service_test.go [hunk 2]
```diff
@@ -1068,9 +1068,284 @@ func TestBlockChildren_SkipSlots(t *testing.T) {
...
+
+	block2 := &pb.BeaconBlock{
+		Slot:             2,
+		ParentRootHash32: root1[:],
+	}
+	err = chainService.beaconDB.SaveBlock(block2)
+	if err != nil {
+		t.Fatalf("Could not save block: %v", err)
+	}
+	err = chainService.beaconDB.UpdateChainHead(block2, state)
```
Comment (IvanTheGreatDev) [inline beacon-chain/blockchain/service_test.go]: Same as above for all of these functions that only return `err`

File: beacon-chain/blockchain/service_test.go [hunk 3]
```diff
@@ -1068,9 +1068,284 @@ func TestBlockChildren_SkipSlots(t *testing.T) {
...
+	// B1  - B3 - B5 (State is slot 10)
+	block1 := &pb.BeaconBlock{
+		Slot:             1,
+		ParentRootHash32: []byte{'A'},
+	}
+	root1, err := hashutil.HashBeaconBlock(block1)
+	if err != nil {
+		t.Fatalf("Could not hash block: %v", err)
+	}
+	err = chainService.beaconDB.SaveBlock(block1)
```
Comment (IvanTheGreatDev) [inline beacon-chain/blockchain/service_test.go]: Same as above for this function as well

File: beacon-chain/blockchain/service_test.go (hunk 1 above)
Comment (terenc3t) [reply beacon-chain/blockchain/service_test.go]: I saved 44 lines ;)

**Pull request merged**